packages/peregrine/lib/List/item.js

```diff
@@ -1,54 +1,64 @@
-import React, { Component } from 'react';
+import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
 
 import fromRenderProp from '../util/fromRenderProp';
 
-class Item extends Component {
-    static propTypes = {
-        classes: PropTypes.shape({
-            root: PropTypes.string
-        }),
-        hasFocus: PropTypes.bool,
-        isSelected: PropTypes.bool,
-        item: PropTypes.any.isRequired,
-        itemIndex: PropTypes.number.isRequired,
-        render: PropTypes.oneOfType([PropTypes.func, PropTypes.string])
-            .isRequired
-    };
-
-    static defaultProps = {
-        classes: {},
-        hasFocus: false,
-        isSelected: false,
-        render: 'div'
-    };
+const isString = item => typeof item === 'string';
 
-    get children() {
-        const { item } = this.props;
-        const isString = typeof item === 'string';
+const getChild = item => (isString(item) ? item : null);
 
-        return isString ? item : null;
-    }
+const Item = props => {
+    const {
+        uniqueID: key,
+        classes,
+        hasFocus,
+        isSelected,
+        item,
+        itemIndex,
+        render,
+        updateSelection,
+        setFocus,
+        ...restProps
+    } = props;
+    const onClick = useCallback(() => updateSelection(key), [
+        updateSelection,
+        key
+    ]);
+    const onFocus = useCallback(() => setFocus(key), [setFocus, key]);
+    const customProps = {
+        classes,
+        hasFocus,
+        isSelected,
+        item,
+        itemIndex,
+        onClick,
+        onFocus
+    };
+    const Root = fromRenderProp(render, Object.keys(customProps));
```

Comment:
In [List](https://github.com/magento/pwa-studio/blob/bcc327e293f3979951d8db5291380aabe7eea0fb/packages/peregrine/lib/List/list.js#L23-L36) you used `useMemo` to prevent having to calculate `customProps` and `Root` each time. Any reason you didn't do that again here?

Reply:
My bad. Missed it. Done.